docker-compose.yml

```diff
@@ -115,13 +118,59 @@ services:
       # - WORKFLOW_TEMPORAL_TLS_KEY
     networks:
       - backend-network
-      - db-connector-network
-      - workflows-network
+      - code-executor-network
+    restart: on-failure
+
+  # Remove if not using Workflows or isolation from main backend serving editor and UI requests is not needed
+  workflows-backend:
+    build:
+      context: ./
+      dockerfile: Dockerfile
+    env_file: ./docker.env
+    environment:
+      - SERVICE_TYPE=WORKFLOW_BACKEND,DB_CONNECTOR,DB_SSH_CONNECTOR
+      - WORKFLOW_BACKEND_HOST=http://workflows-backend:3000
+      - CODE_EXECUTOR_INGRESS_DOMAIN=http://code-executor:3004
+    networks:
+      - backend-network
+      - code-executor-network
+    depends_on:
+      - postgres
+      - retooldb-postgres
+      - workflows-worker
+      - code-executor
+    command: bash -c "./docker_scripts/wait-for-it.sh postgres:5432; ./docker_scripts/start_api.sh"
+    links:
+      - postgres
+    restart: on-failure
+    volumes:
+      - ./keys:/root/.ssh
+      - ssh:/retool_backend/autogen_ssh_keys
+      - ./retool:/usr/local/retool-git-repo
+      - ${BOOTSTRAP_SOURCE:-./retool}:/usr/local/retool-repo
+
+  # Remove this if you are not using the code executor service
+  code-executor:
+    build:
+      context: ./
+      dockerfile: CodeExecutor.Dockerfile
+    command: bash -c "./start.sh"
+    env_file: ./docker.env
+    environment:
+      - NODE_OPTIONS=--max_old_space_size=1024
+    networks:
+      - code-executor-network
+    # code-executor uses nsjail to sandbox code execution. nsjail requires
+    # privileged container access.
+    # If your deployment does not support privileged access, you can set this
+    # to false to not use nsjail. Without nsjail, all code is run without
+    # sandboxing within your deployment.
+    privileged: true
```

Comment:
consider this as default off? otherwise may be a friction point for generally running Retool